src/CRVBadDebtRepayment.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+import {AaveV2Ethereum} from "@aave-address-book/AaveV2Ethereum.sol";
+import {AggregatorV3Interface} from "./external/AggregatorV3Interface.sol";
+import {IERC20} from "@openzeppelin/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/token/ERC20/utils/SafeERC20.sol";
+
+/// @title CRVBadDebtRepayment
+/// @author Llama
+/// @notice Contract to purchase USDC/aUSDC with CRV. Max of 2M USDC/aUSDC sold, max of 2.65M CRV in
+contract CRVBadDebtRepayment {
+    using SafeERC20 for IERC20;
+
+    uint256 public constant AUSD_CAP = 2_000_000e6;
+    uint256 public constant CRV_CAP = 2_656_500e18;
+    uint256 public totalCRVReceived;
+    uint256 public totalAUSDCSold;
+
+    IERC20 public constant CRV = IERC20(0xD533a949740bb3306d119CC777fa900bA034cd52);
+    IERC20 public constant USDC = IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    IERC20 public constant AUSDC = IERC20(0xBcca60bB61934080951369a648Fb03DF4F96263C);
+
+    AggregatorV3Interface public constant CRV_USD_FEED =
+        AggregatorV3Interface(0xCd627aA160A6fA45Eb793D19Ef54f5062F20f33f);
+
+    address public constant BAD_DEBTOR = 0x57E04786E231Af3343562C062E0d058F25daCE9E;
+
+    event Purchase(address indexed tokenIn, address indexed tokenOut, uint256 amountIn, uint256 amountOut);
+
+    /// CRV cap fulfilled
+    error ExcessCRVAmountIn(uint256 amountLeft);
+    /// AUSDC cap fulfilled
+    error ExcessAUSDCPurchased(uint256 amountLeft);
+    /// Not enough CRV in contract to repay bad debt
+    error NotEnoughCRV(uint256 amount);
+    /// Oracle price is 0 or lower
+    error InvalidOracleAnswer();
+    /// Need to request more than 0 tokens out
+    error OnlyNonZeroAmount();
+
+    /// @notice Purchase USDC for CRV
+    /// @param amountIn Amount of BAL input
+    /// @param toUnderlying Whether to receive as USDC (true) or aUSDC (false)
+    /// @return amountOut Amount of USDC received
+    /// @dev Purchaser has to approve BAL transfer before calling this function
+    function purchase(uint256 amountIn, bool toUnderlying) external returns (uint256 amountOut) {
+        if (amountIn == 0) revert OnlyNonZeroAmount();
+        if (amountIn > availableCRVToBeFilled()) revert ExcessCRVAmountIn(availableCRVToBeFilled());
+
+        amountOut = getAmountOut(amountIn);
+        if (amountOut == 0) revert OnlyNonZeroAmount();
+        if (amountOut > availableAUSDCToBeSold()) revert ExcessAUSDCPurchased(availableAUSDCToBeSold());
+
+        totalCRVReceived += amountIn;
+        totalAUSDCSold += amountOut;
+
+        CRV.safeTransferFrom(msg.sender, address(this), amountIn);
```

Comment:
I would highly recommend not holding any asset in this contract. So just transfer the CRV to the AAVE Collector. As for the repay function, you can just approve this contract for the debt amount as part of the proposal payload.